From: Julian Andres Klode <julian.klode@canonical.com>
Date: Fri, 19 Jun 2020 12:57:19 +0200
Subject: Pass dis_ucode_ldr to kernel for recovery mode

In case of a botched microcode update, this allows people to
easily roll back.

It will of course break in the more unlikely event that you are
missing a microcode update in your firmware that is needed to boot
the system, but editing the entry to remove an option is easier than
having to figure out the option and add it.

LP: #1831789
---
 util/grub.d/10_linux.in | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/util/grub.d/10_linux.in b/util/grub.d/10_linux.in
index 260e857..4adb51d 100644
--- a/util/grub.d/10_linux.in
+++ b/util/grub.d/10_linux.in
@@ -228,6 +228,10 @@ case "$machine" in
     *) GENKERNEL_ARCH="$machine" ;;
 esac
 
+case "$GENKERNEL_ARCH" in
+  x86*) GRUB_CMDLINE_LINUX_RECOVERY="$GRUB_CMDLINE_LINUX_RECOVERY dis_ucode_ldr";;
+esac
+
 prepare_boot_cache=
 prepare_root_cache=
 boot_device_id=
